torch/_export/non_strict_utils.py

```diff
@@ -193,6 +198,18 @@ def make_constraints(
     #   - eval_frame.py solves constraints
     #   - _trace.py installs shape metadata in IR.
 
+    if dynamic_shapes is None:
+        return {}
+
+    flat_dynamic_shapes, _ = tree_flatten(
+        dynamic_shapes,
+        is_leaf=lambda x: x is None
+        or all(
+            not isinstance(y, Iterable)
```

Comment:
you mean `isinstance(y, _Dim)` instead, no?

Reply:
Basically IIUC you're trying to keep things like `None` and `{0: dim}` as leaves.

Reply:
Actually look at `update_symbols` in `process_dynamic_shapes` for the patterns here. Another one is `(dim1, dim2)` or `[dim1, dim2]`.

Reply:
@avikchaudhuri Hmm I think the best I can do is 
```
is_leaf=lambda x: x is None
    or all(
        isinstance(y, _Dim) or y is None
        for y in (x.values() if isinstance(x, dict) else x)
    )
```

Because I think we still want to support things like `({0: dim, 1: None}, None)`

Reply:
Well tuples where everything is dim or None is also in scope for leaves, right?
So either None, or a dict where all values are dim or None, or a tuple where everything is dim or None?

Correct me if I'm wrong. Basically {0: dim, 1: None} can be alternatively written as (dim, None) I think. None can mean static dim or None can mean, this tensor is entirely made of static dims.

Reply:
Btw if the lambda is getting too complicated lol pull it out as a function.